File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        // return the message
        long unsigned int start{line.find(" ") + 1};
        return line.substr(start);
    }

    std::string log_level(std::string line) {
        // return the log level
        long unsigned int start{line.find("[") + 1};
        long unsigned int stop{line.find("]") - 1};
        return line.substr(start, stop);
    }

    std::string reformat(std::string line) {
        // return the reformatted message
        return message(line) + " (" + log_level(line) + ")";
    }
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
import java.util.*;
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class LogLevels {
    
    public static String message(String logLine) {
        int startIndex = logLine.indexOf(" ") + 1;
        return logLine.substring(startIndex).trim();
    }

    public static String logLevel(String logLine) {
        int endIndex = logLine.indexOf("]");
        return logLine.substring(1, endIndex).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
